src/ts/ConvivaAnalytics.ts

```diff
@@ -174,35 +185,42 @@ export class ConvivaAnalytics {
     this.debugLog('startsession', this.sessionKey);
   }
 
-  private updateSession = () => {
-    if (!this.sessionDataPopulated) {
-      this.sessionDataPopulated = true;
-      this.updateContentMetadata();
-      this.playerStateManager.updateContentMetadata(this.contentMetadata);
-    }
-  };
-
-  private updateContentMetadata() {
+  /**
+   * Update contentMetadata which must be present before first video frame
+   */
+  private buildContentMetadata() {
     let source = this.player.getConfig().source;
 
-    if (this.contentMetadata.assetName !== this.getAssetName(source)) {
-      this.contentMetadata.assetName = this.getAssetName(source);
-    }
-
+    this.contentMetadata.applicationName = this.config.applicationName || 'Unknown (no config.applicationName set)';
+    this.contentMetadata.assetName = this.getAssetName(source);
     this.contentMetadata.viewerId = source.viewerId || this.config.viewerId || null;
-    this.contentMetadata.streamUrl = this.getUrlFromSource(source);
+    this.contentMetadata.duration = this.player.getDuration();
+    this.contentMetadata.streamType =
+      this.player.isLive() ? Conviva.ContentMetadata.StreamType.LIVE : Conviva.ContentMetadata.StreamType.VOD;
+
     this.contentMetadata.custom = {
       'playerType': this.player.getPlayerType(),
       'streamType': this.player.getStreamType(),
       'vrContentType': this.player.getVRStatus().contentType,
       ...this.config.customTags,
     };
-    this.contentMetadata.duration = this.player.getDuration();
-    this.contentMetadata.streamType
-      = this.player.isLive() ? Conviva.ContentMetadata.StreamType.LIVE : Conviva.ContentMetadata.StreamType.VOD;
+  }
+
+  /**
+   * Update contentMetadata which are allowed during the session
+   */
+  private updateSession() {
+    if (!this.isValidSession()) {
+      return;
+    }
+    this.contentMetadata.streamUrl = this.getUrlFromSource(this.player.getConfig().source); // TODO: mid Meta data available
```

Comment:
What does the TODO mean?

Reply:
Was a note for myself. Forgot to remove the TODO.